docs/source/dataset_vs_iterable_dataset.mdx

```diff
@@ -0,0 +1,217 @@
+
+## Downloading and streaming
+
+When you have a regular "map-style" [`Dataset`], you can access it using `my_dataset[0]`: we have what we call "random access" to the rows.
+For example you can download ImageNet-1k like this:
+
+```python
+from datasets import load_dataset
+
+imagenet = load_dataset("imagenet-1k", split="train")  # downloads the full dataset
+print(imagenet[0])
+```
+
+But one caveat is that you must have all the dataset stored on your disk or in memory, which makes it not always suitable for datasets bigger than disk.
+Because it can become inconvenient for big datasets, there exists another type of dataset: [`IterableDataset`].
+When you have an iterable dataset, you can access it using a `for` loop to load the data progressively as you iterate over the dataset.
+This way, only a small fraction of examples is loaded in memory, and you don't write anything on disk.
+
+For example, you can stream the ImageNet-1k dataset without downloading it completely:
+
+```python
+from datasets import load_dataset
+
+imagenet = load_dataset("imagenet-1k", split="train", streaming=True)  # will start loading the data when iterated over
+for example in imagenet:
+    print(example)
+    break
+```
+
+Dataset streaming can be used to read online data without writing any file on disk.
+For example you can stream datasets made out of multiple hundreds of gigabytes like [C4](https://huggingface.co/datasets/c4), [OSCAR](https://huggingface.co/datasets/oscar) or [LAION-2B](https://huggingface.co/datasets/laion/laion2B-en).
+Learn more about dataset streaming in the [Dataset Streaming Guide](./stream).
+
+This is not the only difference.
+This "lazy" behavior of iterable datasets is also present when it comes to dataset creation and processing.
+
+## Creating map-style datasets and iterable datasets
+
+You can create a [`Dataset`] using lists or dictionaries, and the data is entirely converted to Arrow so you can access the rows easily
+```python
+my_dataset = Dataset.from_dict({"col_1": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]})
+print(my_dataset[0])
+```
+
+To create an iterable dataset on the other hand, you must provide a "lazy" way to load the data.
+In Python, we generally use generator functions. These functions `yield` one example at a time:
+```python
+def my_generator(n):
+    for i in range(n):
+        yield {"col_1": i}
+
+my_iterable_dataset = IterableDataset.from_generator(my_generator, gen_kwargs={"n": 10})
+for example in my_iterable_dataset:
+    print(example)
+    break
+```
+
+## Loading local files entirely and progressively
+
+It is possible to convert local or remote data files to an Arrow [`Dataset`] using [`load_dataset`]:
+```python
+data_files = {"train": ["path/to/data.csv"]}
+my_dataset = load_dataset("csv", data_files=data_files, split="train")
+print(my_dataset[0])
+```
+
+To not have to wait for the conversion to Arrow, you can define an iterable dataset by streaming from your local files.
```

Comment:
What is the benefit of not converting Dataset to Arrow (obvs it’s faster, but it’d be good to mention this explicitly for the user)?

Reply:
Faster + save disk space + you can modify your original data and re-instantiate the dataset without having to reconvert the original data. I'll mention this !